core/src/main/java/org/infinispan/commands/write/BackupWriteCommand.java

```diff
@@ -0,0 +1,297 @@
+   }
+
+   private BackupWriteCommand(Operation operation, CommandInvocationId commandInvocationId, Object key, Object value,
+                              Metadata metadata, long flags, int topologyId) {
+      this.operation = operation;
+      this.commandInvocationId = commandInvocationId;
+      this.key = key;
+      this.value = value;
+      this.metadata = metadata;
+      this.setFlagsBitSet(flags);
+      this.topologyId = topologyId;
+   }
+
+   private static void unRemoveEntry(MVCCEntry<?, ?> e) {
+      e.setCreated(true);
+      e.setExpired(false);
+      e.setRemoved(false);
+      e.setValid(true);
+   }
+
+   private static Operation valueOf(int index) {
+      return CACHED_VALUES[index];
+   }
+
+   public static BackupWriteCommand constructWrite(CommandInvocationId id, Object key, Object value, Metadata metadata, long flags, int topologyId) {
+      return new BackupWriteCommand(Operation.WRITE, id, key, value, metadata, flags, topologyId);
+   }
+
+   public static BackupWriteCommand constructRemove(CommandInvocationId id, Object key, long flags, int topologyId) {
+      return new BackupWriteCommand(Operation.REMOVE, id, key, null, null, flags, topologyId);
+   }
+
+   public static BackupWriteCommand constructRemoveExpired(CommandInvocationId id, Object key, long flags, int topologyId) {
+      return new BackupWriteCommand(Operation.REMOVE_EXPIRED, id, key, null, null, flags, topologyId);
+   }
+
+   public void setNotifier(CacheNotifier<Object, Object> notifier) {
+      this.notifier = notifier;
+   }
+
+   public Object getKey() {
+      return key;
+   }
+
+   @Override
+   public Object perform(InvocationContext ctx) throws Throwable {
+      //noinspection unchecked
+      MVCCEntry<Object, Object> e = (MVCCEntry<Object, Object>) ctx.lookupEntry(key);
+
+      if (trace) {
+         log.tracef("Perform backup operation (%s) for key '%s'. Current Entry='%s'.",
+                    operation, key, e);
+      }
+      if (e == null) {
+         return null; //non owner
+      }
+
+      switch (operation) {
+         case WRITE:
+            performPut(e, ctx);
+            break;
+         case REMOVE:
+            performRemove(e, ctx);
+            break;
+         case REMOVE_EXPIRED:
+            performExpiration(e);
+            break;
+      }
+
+      if (trace) {
+         log.tracef("Perform backup operation (%s) for key '%s'. Updated Entry='%s'.",
+                    operation, key, e);
+      }
+      return null;
+   }
+
+   @Override
+   public Object acceptVisitor(InvocationContext ctx, Visitor visitor) throws Throwable {
+      return visitor.visitBackupWriteCommand(ctx, this);
+   }
+
+   @Override
+   public boolean shouldInvoke(InvocationContext ctx) {
+      return true;
+   }
+
+   @Override
+   public boolean ignoreCommandOnStatus(ComponentStatus status) {
+      return false;
+   }
+
+   @Override
+   public boolean readsExistingValues() {
+      return false;
```

Comment:
OK, forget about it... It does not break any tests and I'll change the method soon anyway.
It's just about the method semantics, and when should the value from persistence be loaded.
:skull: to DeltaAware...